doc/JSON-RPC-interface.md

```diff
@@ -5,6 +5,79 @@ The headless daemon `bitcoind` has the JSON-RPC API enabled by default, the GUI
+certain transactions have multiple confirmations even when they aren't
+part of the best block chain.  For this reason, you should not use
+Bitcoin Core for security sensitive operations on systems you do not
+exclusively control, such as shared computers or virtual private
+servers.
+
+By default, the RPC interface can only be accessed by a client running
+on the same computer and only after the client provides a valid
+authentication credential (username and passphrase).  Any program on
+your computer with access to the file system and local network can
+obtain this level of access.  Additionally, other programs on your
+computer can attempt to provide an RPC interface on the same port as
+used by Bitcoin Core in order to trick you into revealing your
+authentication credentials.  For this reason, it is important to only
+use Bitcoin Core for security-sensitive operations on a computer whose
+other programs you trust.
+
+You may optionally allow other computers to remotely control Bitcoin
+Core by setting the `rpcallowip` and `rpcbind` configuration parameters.
+**Do not enable RPC connections over the public Internet.**  These
+settings are only meant for enabling connections over secure private
+networks.  Although Bitcoin Core's RPC interface does use
+authentication, it does not use encryption, so your login credentials
+are sent as clear text that can be read by anyone on your network path.
+Additionally, the RPC interface has not been hardened to withstand
+arbitrary Internet traffic, so changing the above settings to expose it
+to the Internet (even using something like a Tor hidden service) could
+expose you to unconsidered vulnerabilities.  See `bitcoind -help` for
+more information about these settings and other settings described in
+this document.
+
+Related, if you use Bitcoin Core inside a Docker container, you may need
+to expose the RPC port to the host system.  The default way to do this
+in Docker also exposes the port to the public Internet.  Instead, expose
+it only on the host system's localhost, for example: `-p 127.0.0.1:8332:8332`
+
+By default, Bitcoin Core generates unique login credentials each time it
+restarts and puts them into a file readable only by the user that
+started Bitcoin Core, allowing any of that user's RPC clients with
+read access to the file to login automatically.  The file is `.cookie`
+in the Bitcoin Core configuration directory, and using these credentials
+is the preferred RPC authentication method.  If you need to generate
+static login credentials for your programs, you can use the script in
+the `share/rpcauth` directory in the Bitcoin Core source tree.  As a
+final fallback, you can directly use manually-chosen `rpcuser` and
+`rpcpassword` configuration parameters---but you must ensure that you
+choose a strong and unique passphrase (and still don't use insecure
+networks, as mentioned above).
+
+The RPC interface does not guarantee any escaping of data beyond what's
```

Comment:
Maybe add sub-section titles; ports, escaping, remote access … so that it's easier to navigate this text for people looking for a specific thing